task/control-plane-agent/src/mgs_common.rs

```diff
@@ -311,3 +382,117 @@ impl From<RotImageDetailsConvert> for RotImageDetails {
         }
     }
 }
+
+#[derive(Copy, Clone)]
+struct RotCabooseReader<'a> {
+    sprot: &'a SpRot,
+    size: u32,
+    slot: SlotId,
+}
+
+#[derive(Copy, Clone, Debug)]
+enum CabooseOrSprotError {
```

Comment:
We discussed this in chat, but to summarize: the reason I didn't want to add a variant `Caboose(CabooseError)` to `SprotError` is because
(1) that locks the `CabooseError` into the SP-RoT versioning scheme, and
(2) that means we have to translate between `CabooseError` and `SpError` / `RotError`

I don't love the number of similar-ish types in this file; however, a generic `SprotErrorOr<T>` didn't have great ergonomics.